### PR TITLE
fix(sdk): use promises consistently in API

### PR DESCRIPTION
**Pull request by crimeminister**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires or is a documentation update
- [ ] Security Fix

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation website
- [ ] I have made corresponding changes to the literal docs
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/com/kubelt/lib/config/system.cljc [hunk 1]
```diff
@@ -41,3 +45,25 @@
...
+
+(defn config->system
+  "Return a system configuration map including defaults where options were
+  not supplied by the user. In any of the supplied values did not
+  conform to the options schema, an error map is returned."
+  [config-map]
+  (kubelt.spec/conform
+   spec.config/optional-sdk-config config-map
+   (let [sdk-config (merge lib.config.default/sdk config-map)]
+     ;; Check that the final options map (defaults combined with
```
Comment (tangrammer) [inline src/main/com/kubelt/lib/config/system.cljc:58]: my impression is that now the code is enough explicit that we could remove internal comments about where it's happening inside the function 🤔

File: src/main/com/kubelt/sdk/v1.cljs [hunk 2]
```diff
@@ -117,15 +108,21 @@
   to be re-instantiated."
   [system]
   {:pre [(map? system)]}
-  (lib.config.sdk/options system))
+  (lib.promise/promise
```
Comment (tangrammer) [inline src/main/com/kubelt/sdk/v1.cljs:111]: why do we need to treat `options` as an async/promise process?
do we plan to add more logic/implementation on that part?

File: src/main/com/kubelt/sdk/v1.cljs [hunk 3]
```diff
@@ -80,34 +66,39 @@
...
-  calling (init) as the system to halt."
+  calling (init) as the system to halt. Returns a promise/future that
+  resolves when the shutdown is complete, or rejects with an error map
+  describing the problem that occurred."
   [system]
   {:pre [(map? system)]}
-  (if-not (lib.error/error? system)
-    (lib.init/halt! system)
-    true))
+  (lib.promise/promise
```
Comment (tangrammer) [inline src/main/com/kubelt/sdk/v1.cljs:86]: same question as https://github.com/kubelt/kubelt/pull/363/files#r862674536

**Review by tangrammer — COMMENTED** [review]
Added one fix/compilation commit a few comments!

File: src/main/com/kubelt/lib/config/system.cljc (hunk 1 above)
Comment (crimeminister) [reply src/main/com/kubelt/lib/config/system.cljc]: Let's take a look as the literate docs site to see how things read.

**Review by tangrammer — APPROVED** [review]

**Review by tangrammer — APPROVED** [review]
LGTM!

**Pull request merged**